File: problems/python/3318.find-x-sum-of-all-k-long-subarrays-i.py
from typing import List


class Solution:
    def findXSum(self, nums: List[int], k: int, x: int) -> List[int]:
        result = []
        n = len(nums)

        # slide through L
        for i in range(n - k + 1):
            l = i
            r = i + k
            count = {}
            for slide in range(l, r):
                num = nums[slide]
                if not count.get(num):
                    count[num] = 1
                else:
                    count[num] += 1

            t = list(count.items())
            t.sort(key=lambda x: (-x[1], -x[0]))

            num = 0
            for it in t[:x]:
                num += it[0] * it[1]

            result.append(num)

        return result


nums = [1, 1, 2, 2, 3, 4, 2, 3]
k = 6
x = 2
s = Solution()
print(s.findXSum(nums, k, 2))

File: cpp/3318.find-x-sum-of-all-k-long-subarrays-i.cpp
#include <map>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> findXSum(vector<int>& nums, int k, int x) {
        int n = nums.size();
        vector<int> result;
        for (int l = 0; l < n-k; l++) {
            // num, count
            map<int, int> count;
            int max = 0;
            for (int i = l; i < l+k; i++) {
                if (!count.count(nums[i]))
                    count[nums[i]] = 1;
                else
                    count[nums[i]]++;
                max = max > count[nums[i]] ? max : count[nums[i]];
            }

            int temp = 0;
            for (auto it : count) {
                if (it.second == max)
                    temp += it.second * it.first;
            }

            result.push_back(temp);
        }

        return result;
    }
};

int main() {
    vector<int> nums = {1,1,2,2,3,4,2,3};
    Solution s;
    s.findXSum(nums, 6, 2);
}
